sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/parserPostgresqlImpls.ftl

```diff
@@ -0,0 +1,268 @@
+          |
+            val = Literal()
+          )
+        |
+          (
+              <SCHEMA> { name = new SqlIdentifier("search_path", getPos()); }
+            |
+              <NAMES> { name = new SqlIdentifier("client_encoding", getPos()); }
+            |
+              <SEED> { name = new SqlIdentifier("seed", getPos()); }
+            |
+              name = CompoundIdentifier()
+              ( <EQ> | <TO> )
+          )
+          val = PostgresqlSqlOptionValues()
+        ){
+          return new SqlSetOption(s.end(val), scope, name, val);
+        }
+
+    |
+        <RESET> {
+            s.add(this);
+        }
+        (
+            name = CompoundIdentifier()
+        |
+            <ALL> {
+                name = new SqlIdentifier(token.image.toUpperCase(Locale.ROOT),
+                    getPos());
+            }
+        )
+        {
+            return new SqlSetOption(s.end(name), scope, name, null);
+        }
+    )
+}
+
+String PostgresqlOptionScope() :
+{
+}
+{
+    ( <LOCAL> | <SESSION> ) { return token.image.toUpperCase(Locale.ROOT); }
+}
+
+SqlNode PostgresqlSqlOptionValues():
+{
+  final List<SqlNode> list;
+  Span s;
+  SqlNode e;
+}
+{
+   e = PostgresqlSqlOptionValue() { s = span(); list = startList(e); }
+   ( <COMMA> e = PostgresqlSqlOptionValue() { list.add(e); } )*
+   { 
+      return list.size() > 1 ? new SqlNodeList(list, s.end(this)) : e; 
+   }
+}
+
+SqlNode PostgresqlSqlOptionValue():
+{
+  final SqlNode val;
+}
+{
+  (
+      val = Default()
+  |
+      val = Literal()
+  |
+      val = SimpleIdentifier()
+  ) {
+    return val;
+  }
+}
+
+/** DISCARD { ALL | PLANS | SEQUENCES | TEMPORARY | TEMP } */
+SqlNode PostgresqlSqlDiscard() :
+{
+  final Span s;
+}
+{
+  { s = span(); }
+   <DISCARD> ( <ALL> | <PLANS> | <SEQUENCES> | <TEMPORARY> | <TEMP> ) {
+      return SqlDiscard.OPERATOR.createCall(s.end(this), new SqlIdentifier(
+         token.image.toUpperCase(Locale.ROOT), getPos()));
+   }
+}
+
+/**
+ * BEGIN [ WORK | TRANSACTION ] [ transaction_mode [, ...] ]
+ * where transaction_mode is one of:
+ * ISOLATION LEVEL { SERIALIZABLE | REPEATABLE READ | READ COMMITTED | READ UNCOMMITTED }
+ * READ WRITE | READ ONLY
+ * [ NOT ] DEFERRABLE
+ */
+SqlNode PostgresqlSqlBegin() :
+{
+  final Span s;
+  SqlNodeList transactionModeList = SqlNodeList.EMPTY;
+}
+{
+  { s = span(); }
+  <BEGIN> [ ( <WORK> | <TRANSACTION> ) ] [transactionModeList = PostgresqlSqlBeginTransactionModeList()] {
+    return SqlBegin.OPERATOR.createCall(s.end(this), (SqlNode) transactionModeList);
+  }
+}
+
+SqlNodeList PostgresqlSqlBeginTransactionModeList():
+{
+  final List<SqlNode> list;
+  Span s;
+  SqlNode e;
+}
+{
+  { s = span(); }
+  e = PostgresqlSqlBeginTransactionMode() { s = span(); list = startList(e); }
+  ( <COMMA> e = PostgresqlSqlBeginTransactionMode() { list.add(e); } )*
+  { return new SqlNodeList(list, s.end(this)); }
+}
+
+SqlNode PostgresqlSqlBeginTransactionMode():
```

Comment:
Why do we need transaction modes in the grammar?

Reply:
I imported everything from Babel, will make it easier to merge future changes.